vcd/datasource_vcd_rde_behavior_invocation.go

```diff
@@ -0,0 +1,66 @@
+package vcd
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// This data source is quite special, as it represents an imperative call to a function (aka invoking a behavior)
+// in the declarative world of Terraform. Despite being a data source, whose goal is to perform Read-only operations, invocations
+// can mutate RDE contents. The nature of this one is similar to the built-in "http" provider, which has the "http_http" data source
+// that can perform "PUT"/"POST"/"DELETE" operations.
+func datasourceVcdRdeBehaviorInvocation() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdRdeBehaviorInvocationRead,
+		Schema: map[string]*schema.Schema{
+			"rde_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of the RDE for which the Behavior will be invoked",
+			},
+			"behavior_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of either a RDE Interface Behavior or RDE Type Behavior to be invoked",
+			},
+			"arguments": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "The arguments to be passed to the invoked Behavior",
+			},
+			"metadata": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "Metadata to be passed to the invoked Behavior",
+			},
+			"result": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The raw result of the Behavior invocation",
+			},
+		},
+	}
+}
+
+func datasourceVcdRdeBehaviorInvocationRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	rdeId := d.Get("rde_id").(string)
+	behaviorId := d.Get("behavior_id").(string)
+
+	rde, err := vcdClient.GetRdeById(rdeId)
+	if err != nil {
+		return diag.Errorf("[RDE Behavior Invocation] could not retrieve the RDE with ID '%s': %s", rdeId, err)
+	}
+	result, err := rde.InvokeBehavior(behaviorId, types.BehaviorInvocation{
+		Arguments: d.Get("arguments").(map[string]interface{}),
+		Metadata:  d.Get("metadata").(map[string]interface{}),
+	})
+	if err != nil {
+		return diag.Errorf("[RDE Behavior Invocation] could not invoke the Behavior of the RDE with ID '%s': %s", rdeId, err)
+	}
+	d.SetId(rdeId + "|" + behaviorId) // Invocations are not real entities, so we make an artificial ID.
```

Comment:
Do we have such ID format (RDE ID + Behavior ID) anywhere else? I'm trying to understand whether this format is unique to invocations or not. Thanks!

Reply:
I don't think so. The ID here is useless, it's more a requirement of Terraform to register and control the entity in the state, so I chose two of the attributes that make an invocation unique: The RDE and the Behavior itself. The `|` was just to choose an "easy to spot" separator.

But I'm open for suggestions, maybe @dataclouder @adezxc or @Didainius can jump in and suggest something else.

Reply:
The two IDs with a separator look good, esp. if no such combination is anywhere else, as that will aid in troubleshooting.

(I will mark this conversation as resolved, so it doesn't linger, but it's open for everyone to chime in/re-open, if needed.)